code/modules/cm_tech/trees/marine.dm

```diff
@@ -0,0 +1,156 @@
+	return !ishuman(H)
+
+/datum/techtree/marine/proc/transfer_leader_to(var/mob/living/carbon/human/H)
+	if(!H)
+		return
+	remove_leader()
+
+	RegisterSignal(H, COMSIG_MOVABLE_MOVED, .proc/handle_zlevel_check)
+	RegisterSignal(H, COMSIG_MOB_DEATH, .proc/handle_death)
+
+	leader = H
+
+/datum/techtree/marine/proc/remove_leader()
+	if(!leader)
+		return
+	UnregisterSignal(leader, list(
+		COMSIG_MOB_DEATH,
+		COMSIG_MOVABLE_MOVED
+	))
+	leader = null
+
+/datum/techtree/marine/proc/handle_death(var/mob/living/carbon/human/H)
+	SIGNAL_HANDLER
+	if((H.job in job_cannot_be_overriden) && (!dead_leader || !dead_leader.check_tod()))
+		RegisterSignal(H, COMSIG_PARENT_QDELETING, .proc/cleanup_dead_leader)
+		RegisterSignal(H, COMSIG_HUMAN_REVIVED, .proc/readd_leader)
+		dead_leader = H
+	remove_leader()
+
+/datum/techtree/marine/proc/cleanup_dead_leader(var/mob/living/carbon/human/H)
+	SIGNAL_HANDLER
+	if(dead_leader == H)
+		dead_leader = null
+
+/datum/techtree/marine/proc/readd_leader(var/mob/living/carbon/human/H)
+	SIGNAL_HANDLER
+	if(H != dead_leader)
+		stack_trace("Non-leader attempted to be re-added back to command.")
+		return
+	remove_dead_leader()
+	transfer_leader_to(H)
+
+/datum/techtree/marine/proc/handle_zlevel_check(var/mob/living/carbon/human/H)
+	SIGNAL_HANDLER
+	if(!is_mainship_level(H.z))
+		remove_leader()
+
+/datum/techtree/marine/proc/remove_dead_leader()
+	if(!dead_leader)
+		return
+
+	UnregisterSignal(dead_leader, list(
+		COMSIG_HUMAN_REVIVED,
+		COMSIG_PARENT_QDELETING
+	))
+	dead_leader = null
+
+GLOBAL_LIST_EMPTY(tech_controls_marine)
+
+/obj/structure/machinery/computer/tech_control
+	name = "tech control console"
+	desc = "A console used to acquire the means to control the techtree."
+
+	icon_state = "techweb"
+
+	req_access = list(ACCESS_MARINE_BRIDGE)
+	density = TRUE
+	anchored = TRUE
+	wrenchable = FALSE
+
+	var/datum/techtree/marine/attached_tree
+
+/obj/structure/machinery/computer/tech_control/Initialize()
+	. = ..()
+	GLOB.tech_controls_marine += src
+	if(SStechtree.initialized)
+		attached_tree = GET_TREE(TREE_MARINE)
+
+/obj/structure/machinery/computer/tech_control/Destroy()
+	GLOB.tech_controls_marine -= src
+	attached_tree = null
+	return ..()
+
+// Disallow deconstructing
+/obj/structure/machinery/computer/tech_control/attackby(obj/item/I, mob/user)
+	return
+
+/obj/structure/machinery/computer/tech_control/attack_hand(var/mob/M)
+	. = ..()
+
+	if(!skillcheck(M, SKILL_INTEL, SKILL_INTEL_TRAINED) && SSmapping.configs[GROUND_MAP].map_name != MAP_WHISKEY_OUTPOST)
+		to_chat(M, SPAN_WARNING("You don't have the training to use [src]."))
+		return
+
+	if(!attached_tree)
+		return
+
+	//M.mind.view_objective_memories(src)
+	var/datum/techtree/tree = GET_TREE(TREE_MARINE)
+	tree.enter_mob(usr, FALSE)
+
+/datum/techtree/marine/on_tier_change(datum/tier/oldtier)
+	if(tier.tier < 2)
+		return //No need to announce tier updates for tier 1
+	var/name = "ALMAYER TECH LEVEL UPGRADED"
```

Comment:
we might want to take a second look at this announcement (Defcon sounds way better imo)